transform_utils.ml

```diff
@@ -19,10 +19,76 @@ let change_instrs (transform : pc -> instruction_change) ({formals; instrs} : an
       | Replace is ->
         acc_instr (pc+1) (List.rev_append is acc) true
       | Insert is ->
+        assert(match[@warning "-4"] instrs.(pc) with | Label _ -> false | _ -> true);
         acc_instr (pc+1) (instrs.(pc) :: List.rev_append is acc) true
       | InsertAfter is ->
         acc_instr (pc+1) (List.rev_append is (instrs.(pc) :: acc)) true
       | Unchanged ->
         acc_instr (pc+1) (instrs.(pc)::acc) changed
   in
   acc_instr 0 [] false
+
+
+(* Splits all edges that have multi-predecessors branching *)
+let normalize_graph instrs =
+  let rec normalize instrs preds pc =
+    let is_branch pred_pc = match[@warning "-4"] instrs.(pred_pc) with
```

Comment:
Formatting nitpick: put the match on a new line and indent it (and also indent the cases).